src/index.ts

```diff
@@ -0,0 +1,10 @@
+import getSdk = require('resin-sdk');
+import settings = require('resin-settings-client');
+
+export = getSdk({
```

Comment:
Yep. In addition to just normal CommonJS there's actually two more:

* `import X = require('x')`/`export = A`/`export MyFunction` - Semantically the same as commonJS (more or less) but with a slightly different syntax to avoid breaking existing unrelated code that happens to use `require()` in non-Node code.
* ES6 syntax (`import { X } from 'y'`/`export default ...`)

ES6 is encouraged, and gives you fully working typing, but can't give you `module.exports` style exports, which is what we want here.

If you do just use CommonJS, without using the special TypeScript syntax then it works fine at runtime, but TypeScript can't know that it's a module import, so can't follow it to pull the types in, and your variable is just `any`-typed.